Tests/Packet++Test/Tests/DoIpTests.cpp

```diff
@@ -0,0 +1,1402 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "Packet.h"
+#include "EthLayer.h"
+#include "IPv4Layer.h"
+#include "TcpLayer.h"
+#include "UdpLayer.h"
+#include "PayloadLayer.h"
+#include "SystemUtils.h"
+#include "PacketUtils.h"
+#include "DeprecationUtils.h"
+#include <memory>
+#include <array>
+#include "DoIpLayer.h"
+#include "GeneralUtils.h"
+
+// RoutingActivationRequest
+PTF_TEST_CASE(DoIpRoutingActivationRequestPacketParsing)
+{
+	// Dissect Routing Activation Request message
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/DoIpRoutingActivationRequestPacket.dat");
+
+	pcpp::Packet RoutingActivationRequest(&rawPacket1);
+	PTF_ASSERT_TRUE(RoutingActivationRequest.isPacketOfType(pcpp::IPv4));
+	PTF_ASSERT_TRUE(RoutingActivationRequest.isPacketOfType(pcpp::TCP));
+	PTF_ASSERT_TRUE(RoutingActivationRequest.isPacketOfType(pcpp::DOIP));
+
+	pcpp::TcpLayer* tcpLayer = RoutingActivationRequest.getLayerOfType<pcpp::TcpLayer>();
+	PTF_ASSERT_NOT_NULL(tcpLayer);
+
+	PTF_ASSERT_EQUAL(tcpLayer->getDstPort(), pcpp::DoIpPorts::TCP_UDP_PORT);
+
+	auto* doipLayer = RoutingActivationRequest.getLayerOfType<pcpp::DoIpRoutingActivationRequest>();
+
+	PTF_ASSERT_NOT_NULL(doipLayer);
+
+	PTF_ASSERT_EQUAL(doipLayer->getProtocolVersion(), pcpp::DoIpProtocolVersion::ISO13400_2012, enumclass);
+	PTF_ASSERT_EQUAL(doipLayer->getInvertProtocolVersion(), 0xFD);
+	PTF_ASSERT_EQUAL(doipLayer->getPayloadType(), pcpp::DoIpPayloadTypes::ROUTING_ACTIVATION_REQUEST, enumclass);
+	PTF_ASSERT_EQUAL(doipLayer->getPayloadTypeAsStr(), "Routing activation request");
+	PTF_ASSERT_EQUAL(doipLayer->getPayloadLength(), 11);
+	PTF_ASSERT_EQUAL(doipLayer->toString(), "DoIP Layer, Routing activation request (0x0005)");
+	PTF_ASSERT_EQUAL(doipLayer->getSourceAddress(), 0x0e80);
+	PTF_ASSERT_EQUAL(doipLayer->getActivationType(), pcpp::DoIpActivationTypes::DEFAULT, enumclass);
+	std::array<uint8_t, 4> isoField = { 0x0, 0x0, 0x0, 0x0 };
+	PTF_ASSERT_VECTORS_EQUAL(doipLayer->getReservedIso(), isoField);
+	PTF_ASSERT_TRUE(doipLayer->hasReservedOem());
+	std::array<uint8_t, 4> oemField = { 0x0, 0x0, 0x0, 0x0 };
+	PTF_ASSERT_BUF_COMPARE(doipLayer->getReservedOem().data(), oemField.data(), 4);
```

Comment:
Let's be consistent and use either `PTF_ASSERT_VECTORS_EQUAL` or `PTF_ASSERT_BUF_COMPARE` in both of these asserts?